File: lib/src/models/product.dart
class Product {
  final String id;
  final String name;
  final String slug;
  final String description;
  final double price;
  final double actualPrice;
  final double marginPrice;
  final int stockQuantity;
  final List<String> images;
  final DateTime createdAt;

  Product({
    required this.id,
    required this.name,
    required this.slug,
    required this.description,
    required this.price,
    this.actualPrice = 0,
    this.marginPrice = 0,
    this.stockQuantity = 0,
    required this.images,
    required this.createdAt,
  });

  factory Product.fromJson(Map<String, dynamic> json) {
    return Product(
      id: json['_id'] ?? json['id'],
      name: json['name'] ?? '',
      slug: json['slug'] ?? '',
      description: json['description'] ?? '',
      price: (json['price'] ?? 0).toDouble(),
      actualPrice: (json['actualPrice'] ?? 0).toDouble(),
      marginPrice: (json['marginPrice'] ?? 0).toDouble(),
      stockQuantity: json['stockQuantity'] ?? 0,
      images: List<String>.from(json['images'] ?? []),
      createdAt: json['createdAt'] != null
          ? DateTime.parse(json['createdAt'])
          : DateTime.now(),
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'name': name,
      'slug': slug,
      'description': description,
      'price': price,
      'actualPrice': actualPrice,
      'marginPrice': marginPrice,
      'stockQuantity': stockQuantity,
      'images': images,
    };
  }
}

File: backend-express/src/models/Product.js
const mongoose = require('mongoose');

const productSchema = new mongoose.Schema({
  name: { type: String, required: true },
  slug: { type: String, required: true, unique: true },
  description: { type: String, default: '' },
  price: { type: Number, required: true, min: 0 },
  actualPrice: { type: Number, default: 0, min: 0 },
  marginPrice: { type: Number, default: 0, min: 0 },
  stockQuantity: { type: Number, default: 0, min: 0 },
  images: { type: [String], default: [] },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('Product', productSchema);

